app/components/course-page/course-stage-step/first-stage-instructions-card.ts

```diff
@@ -35,7 +35,7 @@ class NavigateToFileStep extends BaseStep implements Step {
       return 'Navigate to README.md';
     }
 
-    const filename = this.repository.firstStageSolution.changedFiles[0].filename;
+    const filename = this.repository.firstStageSolution?.changedFiles[0]!.filename;
```

Comment:
```suggestion
    const filename = this.repository.firstStageSolution.changedFiles[0]!.filename;
```

(Shouldn't this work? `this.repository.firstStageSolution` was already part of an if condition above, so it shouldn't be null at this point)